service/history/workflow/update/update.go

```diff
@@ -351,6 +356,31 @@ func (u *Update) onResponseMsg(
 	return nil
 }
 
+func (u *Update) cancel(
+	_ context.Context,
+	eventStore EventStore,
+) {
+	const waitAccepted = stateAdmitted | stateProvisionallyRequested | stateRequested
+	const waitCompleted = stateAccepted | stateProvisionallyAccepted
+
+	prevState := u.setState(stateProvisionallyCompleted)
+	eventStore.OnAfterCommit(func(context.Context) {
+		if prevState.Matches(stateSet(waitAccepted)) {
+			u.request = nil
+			u.setState(stateCompleted)
+			u.accepted.(*future.FutureImpl[*failurepb.Failure]).Set(nil, CancelErr)
+			u.outcome.(*future.FutureImpl[*updatepb.Outcome]).Set(nil, CancelErr)
```

Comment:
Can this call get mixed in with WFT processing? Is it possible that these futures have already been Set here? prevState was observed _before_ the effect commit starts cascading through